### DISCUSSION TITLE
Is there another way to connect with PLC such as using profibus address or MPI

### DISCUSSION THREAD

**Discussion by ilmi** [gijzelaerr/python-snap7#441] [Q&A] [answered]
Hello..,, currently my plc connected to PC via profibus, is it possible to access the PLC data if I installed python and snap7 lib inside the PC. If possible how should I setup the connection.

I already try to setup the connection like this :

profibus_address = 2
baud_rate = 187500
timeout = 1000

plc = snap7.client.Client()
plc.connect('PROFIBUS',profibus_address, baud_rate, timeout)

but I still get the error. 

Thanks for you're help

**Comment by nikteliy** [comment, marked as answer]
Hi @ilmi 
`Snap7` does not support `profibus`. It only supports `s7comm` over `tcp`
Let me guess, ChatGPT gave you this code? :)

![NET_S7_Protokoll_01](https://user-images.githubusercontent.com/52915342/230323258-962487d0-342c-4371-bdb7-24d8e7ca0a73.png)

By the way, take a look at the [pyprofibus](https://github.com/mbuesch/pyprofibus) library

**Comment by gijzelaerr** [comment]
Closing as answered. Snap7 only supports S7comm over TCP/IP. It does not support Profibus or MPI.
